### PR TITLE
Replace jsonschema with a tiny piece of validation logic

### PR DESCRIPTION
**Pull request by mqp**
This was a pretty heavy dependency (a few dozen minified kilobytes) so it's unreasonable to take it and then barely use any of its features.

(It was actually even bigger than it looked, because it also secretly depended on Webpack polyfilling the Node.js `url` standard library module.)

### CODE REVIEW

**Comment by gfodor** [comment]
This seems like the wrong trade-off to me. The downside here is now we are exposed to bugs or regressions in the new validation logic. If there were a bunch of unit tests to ensure we are not going to get burned by this I'd feel better, but it certainly doesn't seem worth a few kilobytes to lose confidence in the schema validation routines.

**Comment by mqp** [comment]
I'm happy to make some unit tests, so if that changes your mind then we agree.

(I didn't know we had a mechanism for tests, but I see that we do now!)

**Comment by gfodor** [comment]
Great thanks!

**Comment by gfodor** [comment]
Also I don't know if we added the test runner to npm build -- can you do that if not?

File: test/unit/utils/validate-storage-schema.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    settings: {},
+    confirmedDiscordRooms: [],
+    uploadPromotionTokens: []
+  }, exampleSchema));
+
+  t.throws(() => validate({
+    profile: { displayName: "John" },
+    settings: { isBot: true },
+    confirmedDiscordRooms: ["general"],
+    uploadPromotionTokens: [{ fileId: "whiz", promotionToken: 94329 }]
```
Comment (johnshaughnessy) [inline test/unit/utils/validate-storage-schema.test.js]: bang

**Comment by mqp** [comment]
> Also I don't know if we added the test runner to npm build -- can you do that if not?

Well, right now, it's rather the opposite -- `npm test` runs the build as a "kind of test". Do you know a whole other way you endorse?

If I were designing it, I would have `test` and `build` be totally separate and neither one does the other.

**Comment by gfodor** [comment]
don't have a strong opinion -- don't know npm culture. in java land,
typically the happy path build script runs the tests for example. in any
case, main thing is to ensure that our actual CI build runs the tests and
fails the build if it fails.

**Comment by netpro2k** [comment]
I would keep it as `npm test`, but I agree our CI scripts should run the tests if they don't already.

**Pull request closed without merging**